app/components/channel_link/channel_link.js

```diff
@@ -39,12 +50,35 @@ export default class ChannelLink extends React.PureComponent {
         return null;
     }
 
-    handlePress = () => {
-        this.props.actions.setChannelDisplayName(this.state.channel.display_name);
-        this.props.actions.handleSelectChannel(this.state.channel.id);
+    handlePress = async () => {
+        const {channel: channelState} = this.state;
```

Comment:
Minor suggestion, but  you could simplify this by using a let and removing `channelState`
```javascript
let {channel} = this.state;
```